core/xml.js

```diff
@@ -311,7 +330,7 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       // to be moved to a nested destination in the next operation.
       var block = Blockly.Xml.domToBlock(xmlChild, workspace);
       var blockX = parseInt(xmlChild.getAttribute('x'), 10);
-      var blockY = parseInt(xmlChild.getAttribute('y'), 10);
+      var blockY = parseInt(xmlChild.getAttribute('y'), 10) + bottomY; // add bottomY to be below all blocks
```

Comment:
Indeed, thanks. 
I now keep track of the block and their position before the move, compute the offset and move the blocks. That way I move the block only once.